### DISCUSSION TITLE
[help] How to use geodata package with targets

### DISCUSSION THREAD

**Discussion by njtierney** [ropensci/targets#1238] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi folks, 

I want to have a target that gets data using the `geodata` package, specifically the `crop_spam()` function - like so:

```
 crop_coffee_data = crop_spam(
    crop = "acof",
    var = "area",
    path = "data",
    africa = TRUE
    )
```

This package requires that you specify a directory to save the data to, but at the time of writing I am not 100% sure what the file names will be. I was wondering what the best practice is to be able to:

- store the output of these downloaded files
- make sure they don't run again when we run `tar_make()`

Let me know if I can make this clearer or provide more information!

Cheers,

Nick

**Comment by Aariq** [comment, marked as answer]
Since it looks like `crop_spam()` returns a SpatRaster object, the discussion [here](https://github.com/ropensci/targets/discussions/809) might be useful.  In particular, I've been using the following for the `format` argument for targets that return SpatRasters with success.

``` r
format_geotiff <- tar_format(
  read = function(path) terra::rast(path),
  write = function(object, path) terra::writeRaster(x = object, filename = path, filetype = "GTiff", overwrite = TRUE),
  marshal = function(object) terra::wrap(object),
  unmarshal = function(object) terra::unwrap(object)
)

list(
tar_target(coffee_data, crop_coffee_data(), format = format_geotiff)
)
```

The other option is to write the results out explicitly as part of your `crop_coffee_data()` function (e.g. with `terra::writeRaster()`) rather than storing the resulting SpatRaster in the _targets store and use `format = "file"`.  You don't need to know the file names ahead of time, you just need a function that returns file names of the written files.  Hope this helps!

**Reply by njtierney** [reply]
Woh that just works! Thanks so much! One thing I wasn't anticipating was that I need to namespace (pkg::fun) the terra functions when using `tar_format`.

I feel like some geospatial targets tarchetype package would be really helpful, my team does this sort of work a lot
